zebra-state/src/service/non_finalized_state.rs

```diff
@@ -90,6 +91,17 @@ impl NonFinalizedState {
 
         // extract best chain
         let mut best_chain = chains.next_back().expect("there's at least one chain");
+
+        // Obtain the treestate associated with the block being finalized.
+        let treestate = Treestate {
+            note_commitment_trees: NoteCommitmentTrees {
+                sprout: best_chain.sprout_note_commitment_tree.clone(),
+                sapling: best_chain.sapling_note_commitment_tree.clone(),
+                orchard: best_chain.orchard_note_commitment_tree.clone(),
+            },
+            history_tree: best_chain.history_tree.clone(),
```

Comment:
These are the trees for the tip block. But we're sending the root block to the finalised state, so we also need to send the root trees. I think this means we need to store a history tree / height index.

It would also be good to move this code into `pop_root()`, which looks up the height of the root block, and drops all the indexes we're not using any more.

Reply:
> These are the trees for the tip block. But we're sending the root block to the finalised state, so we also need to send the root trees.

I think you're right!